File: app/src/main/java/com/example/hurshi/androidtest/MainActivity.java
package com.example.hurshi.androidtest;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        toOtherActivity(R.id.button1, DisabledViewClickEventActivity.class);
    }

    private void toOtherActivity(int resID, final Class<?> cls) {
        findViewById(resID).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Utils.toActivity(MainActivity.this, cls);
            }
        });
    }
}

File: CoordinateLayoutBehaviors/app/src/main/java/com/example/coordinatelayoutbehaviors/MainActivity.kt
package com.example.coordinatelayoutbehaviors

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.coordinatelayoutbehaviors.doublescroll.DoubleScrollActivity
import com.example.coordinatelayoutbehaviors.themirrordraw.TheMirrorDrawActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        clickToActivity(b1, ScrollingActivity::class.java)
        clickToActivity(b2, DoubleScrollActivity::class.java)
        clickToActivity(b3, TheMirrorDrawActivity::class.java)
    }


    private fun clickToActivity(v: View, cls: Class<*>) {
        v.setOnClickListener {
            startActivity(Intent(this@MainActivity, cls))
        }
    }
}
